src/classes/STG_SettingsManager_TEST.cls

```diff
@@ -769,6 +769,424 @@ private with sharing class STG_SettingsManager_TEST {
             'Installment Auto Creation Setting should be correctly converted to respected custom label');
     }
 
+    /**
+    * @description Should prevent read only user to delete opportunity naming settings
+    * as they do not have delete permission
+    */
+    @isTest
+    private static void shouldPreventDeleteOppNamingWhenUserDoesNotHaveDeletePermission() {
+        Opportunity_Naming_Settings__c settings = getOppNameSettings();
+        insert settings;
+
+        Test.startTest();
+
+        System.runAs(buildReadOnlyUser()) {
+            Test.setCurrentPage(Page.STG_PanelOppNaming);
+            STG_PanelOppNaming_CTRL panel = new STG_PanelOppNaming_CTRL();
+            panel.currentONS = settings;
+            panel.currentONSName = settings.Name;
+
+            panel.delONS();
+
+            UTIL_UnitTestData_TEST.assertPageHasError(System.Label.stgDeletePermissionError);
+            System.assertEquals(1, panel.listOppNameSettings.size());
+            System.assertEquals(1, panel.mapOppNameSettings.size());
+            System.assert([select id, name from Opportunity_Naming_Settings__c].size() > 0);
```

Comment:
Instead using the query for custom settings (formatting is off), we can use a helper method

```

    private static void assertCustomSettingsSize(SObjectType sObjType, Integer expectedSize) {
        String soql = new UTIL_Query()
            .withFrom(sObjType)
            .withSelectFields(new String[]{ 'Id', 'Name' })
            .build();

        List<SObject> customSettings = Database.query(soql);

        System.assertEquals(expectedSize, customSettings.size(), 
            'Should return expected custom settings: ' + customSettings);
    }
```

and line 794 would be `assertCustomSettingsSize(Opportunity_Naming_Settings__c.SObjectType, 1);`

Similarly, line 849 would be `assertCustomSettingsSize(npe03__Custom_Field_Mapping__c.SObjectType, 1);` etc.

Reply:
@RadSFDeveloper , Instead of querying the custom settings, I am going to change the assert as : 
System.assertEquals(0, Opportunity_Naming_Settings__c.getAll().size());